ci/run_clang_tidy.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create clang_tidy conda environment"
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --output conda \
+  --file_key clang_tidy \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee env.yaml
+
+rapids-mamba-retry env create --force -f env.yaml -n clang_tidy
+# Temporarily allow unbound variables for conda activation.
+set +u && conda activate clang_tidy && set -u
+
+FORMAT_FILE_URL=https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-23.04/cmake-format-rapids-cmake.json
+export RAPIDS_CMAKE_FORMAT_FILE=/tmp/rapids_cmake_ci/cmake-formats-rapids-cmake.json
+mkdir -p $(dirname ${RAPIDS_CMAKE_FORMAT_FILE})
+wget -O ${RAPIDS_CMAKE_FORMAT_FILE} ${FORMAT_FILE_URL}
+
+export LD_LIBRARY_PATH="$CONDA_PREFIX/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}"
+mkdir -p cpp/build
+cd cpp/build
+cmake -DGPU_ARCHS=70 \
+      -DBLAS_LIBRARIES=${CONDA_PREFIX}/lib/libopenblas.so.0 \
+      ..
+make -j treelite
```

Comment:
why do we need to build treelite?

Reply:
I had the same question for @bdice who supposed that it is likely to put some header files into place. However, I don't understand well enough how clang-tidy works to have a better explanation.

This was changed from `benchmark` to `treelite` in https://github.com/rapidsai/cuml/commit/52320b06ca929db81a02cf7625b881b30f12228a .

Reply:
So, no, we do not actually build anything to run clang-tidy, but we do have to invoke CMake to create the compile command database. That's why I added a `--nobuild` option to our build script in https://github.com/rapidsai/cuml/pull/5235/commits/2c2899510a44a7758c06556e947737b615e59b29 . I assume that the previous targets where an attempt at reducing the build time, but I think not building anything is the better (and faster) approach.